packages/@react-aria/combobox/src/useComboBox.ts

```diff
@@ -0,0 +1,222 @@
+import {useMenuTrigger} from '@react-aria/menu';
+import {useSelectableCollection} from '@react-aria/selection';
+import {useTextField} from '@react-aria/textfield';
+
+interface ComboBoxProps {
+  completionMode?: 'suggest' | 'complete',
+  menuTrigger?: 'focus' | 'input' | 'manual'
+}
+
+interface AriaComboBoxProps<T> extends ComboBoxProps {
+  popoverRef: RefObject<HTMLDivElement>,
+  triggerRef: RefObject<HTMLElement>,
+  inputRef: RefObject<HTMLInputElement>,
+  layout: ListLayout<T>,
+  onBlur?: (e: FocusEvent<Element>) => void, // don't think these two (blur/focus) should be added?
+  onFocus?: (e: FocusEvent<Element>) => void,
+  isFocused: boolean,
+  setIsFocused: (e: boolean) => void
+}
+
+interface ComboBoxAria {
+  triggerProps: HTMLAttributes<HTMLElement> & PressProps,
+  inputProps: HTMLAttributes<HTMLElement>,
+  menuProps: HTMLAttributes<HTMLElement>,
+  labelProps: HTMLAttributes<HTMLElement>
+}
+
+export function useComboBox<T>(props: AriaComboBoxProps<T>, state: ComboBoxState<T>): ComboBoxAria {
+  let {
+    triggerRef,
+    popoverRef,
+    inputRef,
+    layout,
+    isFocused,
+    setIsFocused,
+    completionMode,
+    menuTrigger
+  } = props;
+
+  let {menuTriggerProps, menuProps} = useMenuTrigger(
+    {
+      ref: triggerRef
+    },
+    state
+  );
+
+  let onChange = (val) => {
+    state.setValue(val);
+
+    // If user deletes entry in textfield, clear combobox selection
+    // Probably needs more conditions here (shouldn't clear in controlled?)
+    // Move into a useEffect?
+    if (val === '' && state.selectedKey) {
+      // don't use clearSelection? For some reason toggleSelection doesn't work
+      state.selectionManager.clearSelection();
+      // state.selectionManager.toggleSelection(state.selectedKey);
+    }
+
+    if (menuTrigger !== 'manual') {
+      state.open();
+    }
+  };
+
+  // TODO: Refine the below, feels weird to have focusedItem and also need to still do state.selectionManger.focusedKey
+  let [focusedKeyId, setFocusedKeyId] = useState(null);
+  let focusedItem = state.selectionManager.focusedKey ? state.collection.getItem(state.selectionManager.focusedKey) : null;
+  useEffect(() => {
+    if (focusedItem) {
+      setFocusedKeyId(`${menuProps.id}-option-${focusedItem.key}`);
+    }
+  }, [state.selectionManager.focusedKey, state.collection, focusedItem, setFocusedKeyId, menuProps.id]);
+
+  // Using layout initiated from ComboBox, generate the keydown handlers for textfield (arrow up/down to navigate through menu when focus in the textfield)
+  let {collectionProps} = useSelectableCollection({
+    selectionManager: state.selectionManager,
+    keyboardDelegate: layout,
+    shouldTypeAhead: false,
+    disallowEmptySelection: true
+  });
+
+  // For textfield specific keydown operations
+  let onKeyDown = (e) => {
+    switch (e.key) {
+      case 'Enter':
+        if (state.isOpen && focusedItem) {
+          state.setSelectedKey(state.selectionManager.focusedKey);
```

Comment:
I don't believe so since the items are only being virtually focused so the typical press event that happens when the user hits Enter isn't happening on the listbox items but instead on the input field (unlike in Picker or Menu).  Commenting out this `setSelectedKey` makes it so the item doesn't get selected when the user hits Enter.